### PR TITLE
Docker - Client and common compose

### PR DESCRIPTION
**Pull request by djavorek**
## Description of Changes
- Added **Dockerfile for client**
  - Multi-stage build
  - using node:18-alpine (node 18.4.0)
  - We are building with _adapter-node_ from now, to be able to run SvelteKit as a node app (see: package.json, svelte.config.js)
- Pulled up **docker-compose into root**, added client, and fixed context for server
- Added dockerignore for both client and server to decrease image size.

- Added **new workflow, which builds 1-1 docker image** for server and client, then publishes them into AWS ECR. (images.yml)
  - Docker build takes a while, eating our CI credits like crazy
    - Build waits pain-point builds to succeed before starts. (Updated the name of those jobs to be more clear)
    - Only runs on master commits.

- **Renovated README.md, added DEV.md** for technical stuff.
  - Shiny new badge into README showing status of the new workflow 

- **Using Vite directly** for building and packaging (see: npm scripts, vite.config.js, svelte.config.js)
  - Required because of https://github.com/sveltejs/kit/issues/5184

### CODE REVIEW

File: .github/workflows/images.yml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        uses: aws-actions/amazon-ecr-login@v1
+
+      - name: Server - Build, tag, and push image to Amazon ECR
+        env:
+          ECR_REGISTRY: ${{ steps.login-ecr.outputs.registry }}
+          ECR_REPOSITORY: rezsi-server
+          IMAGE_TAG: latest
+        run: |
+          docker build -t $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG ./server
+          docker push $ECR_REGISTRY/$ECR_REPOSITORY:$IMAGE_TAG
```
Comment (szattila98) [inline .github/workflows/images.yml:77]: This is kinda sick, ngl

File: DEV.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Backend dev setup
+## Prerequisites 
+ - rustup
```
Comment (szattila98) [inline DEV.md:3]: Backend versions should also be specified, I will include it in my next pr.

File: README.md [hunk 3]
```diff
@@ -1,16 +1,9 @@
-# veryrezsi
+# 💲 VeryRezsi 💲
+###### Status
+![Docker Build](https://github.com/szattila98/veryrezsi/actions/workflows/images.yml/badge.svg)
```
Comment (szattila98) [inline README.md:3]: Hella cool, this updates itself based on the statuses I presume.

File: client/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Stage BUILD
+FROM node:18-alpine
```
Comment (szattila98) [inline client/Dockerfile:2]: We should also run the client CI/CD with node 18 then, if possible. And don't forget that in the `images.yml` there is a reference to 16.x, change that as well.

**Review by szattila98 — APPROVED** [review]
Apart from the little nitpick node version, very nice work in troubling times, thank you!

File: client/Dockerfile (hunk 4 above)
Comment (djavorek) [reply client/Dockerfile:2]: Exactly, will do that here.

File: README.md (hunk 3 above)
Comment (djavorek) [reply README.md:3]: Sure it does. We need a lot of badges to be real GH users.

File: client/vite.config.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { sveltekit } from '@sveltejs/kit/vite';
+import { resolve } from 'path';
+
+/** @type {import('vite').UserConfig} */
```
[the remark this answers is not in the document]
Comment (djavorek) [reply client/vite.config.js:4]: Created: https://github.com/szattila98/veryrezsi/issues/33

**Review by szattila98 — APPROVED** [review]

**Pull request merged**